squash/dashboard/viz/AMx/main.py

```diff
@@ -18,35 +18,41 @@
 from api_helper import get_url_args, get_data_as_pandas_df # noqa
 from bokeh_helper import add_span_annotation # noqa
 
-# App name
-BOKEH_APP = 'AMx'
+# Get url query args
+args = get_url_args(curdoc, defaults={'metric': 'AM1'})
+
+
+# App title
+title = Div(text="""<h2>{metric} diagnostic plot for {ci_dataset} dataset from
+                    job ID {ci_id}</h2>""".format_map(args))
 
 # Get data
-args = get_url_args(curdoc, defaults={'metric': 'AM1'})
-data = get_data_as_pandas_df(endpoint=BOKEH_APP,
+data = get_data_as_pandas_df(endpoint='apps',
                              params=args)
 
+
 # Configure bokeh data sources with the full and
-# selected datasets
-snr = data['matchedDataset']['snr']
-dist = data['matchedDataset']['dist']
-full = ColumnDataSource(data={'snr': snr['value'], 'dist': dist['value']})
+# selected data sets
 
-# Selected dataset
-# TODO: Use pandas notation here
 
-index = np.array(snr['value']) > float(args['snr_cut'])
+snr = {'value': [], 'label': '', 'unit': ''}
+selected_snr = []
 
-selected_snr = np.array(snr['value'])[index]
-selected_dist = np.array(dist['value'])[index]
-selected = ColumnDataSource(data={'snr': selected_snr,
-                                  'dist': selected_dist})
+dist = {'value': [], 'label': '', 'unit': ''}
+selected_dist = []
 
-# Configure bokeh widgets
+if not data.empty:
+    snr = data['matchedDataset']['snr']
+    index = np.array(snr['value']) > float(args['snr_cut'])
```

Comment:
I'd keep the np nonsense modular. Just because there is so little of it in an otherwise mostly web based project.